spec/javascripts/spec/feedback.spec.js

```diff
@@ -0,0 +1,101 @@
+describe('Feedback', function () {
+  'use strict'
+
+  it('can extract a path from a URL', function () {
+    expect(GOVUK.feedback.getPathFor('http://example.com/this/is/the/path')).toBe('/this/is/the/path')
+  })
+
+  describe('when on the contact page', function () {
+    var locationPathnameSpy
+
+    beforeEach(function () {
+      locationPathnameSpy = spyOn(GOVUK.feedback, 'getLocationPathname').and.returnValue('/contact')
+      GOVUK.feedback.init()
+    })
+
+    afterEach(function () {
+      locationPathnameSpy.and.callThrough()
+    })
+
+    it('stores the referrer in a cookie', function () {
+      expect(GOVUK.getCookie('govuk_contact_referrer')).toBe(document.referrer)
+    })
+  })
+
+  describe('when there is a feedback form on the page', function () {
+    var contactForm
+
+    beforeAll(function () {
+      contactForm = document.createElement('form')
+      contactForm.setAttribute('class', 'contact-form')
+      var link = document.createElement('input')
+      link.setAttribute('name', 'contact[link]')
+      contactForm.appendChild(link)
+      contactForm.link = link
```

Comment:
It took me a while to figure out what was going on here - "why does a `<form>` need a `link` attribute?".

Rather than setting proprietary properties on the otherwise 'pure' [Element](https://developer.mozilla.org/en-US/docs/Web/API/Element) `contactForm` object, I'd suggest declaring additional variables in the same way as you've declared `contactForm` on line 26, e.g. `var contactFormLink`.

Same goes for `specificLocation`, `javascriptEnabled` and `referrer`.

Reply:
I was going for a nice sounding `expect(contactForm.link).toBe(something)`, but I agree, that's no excuse to confuse things by adding properties to the form object.